can/interfaces/vector/canlib.py

```diff
@@ -363,22 +385,26 @@ def reset(self):
     @staticmethod
     def _detect_available_configs():
         configs = []
-        if vxlapi is None:
-            return configs
-        driver_config = vxlapi.XLdriverConfig()
-        try:
-            vxlapi.xlOpenDriver()
-            vxlapi.xlGetDriverConfig(driver_config)
-            vxlapi.xlCloseDriver()
-        except:
-            pass
-        LOG.info('Found %d channels', driver_config.channelCount)
-        for i in range(driver_config.channelCount):
-            channel_config = driver_config.channel[i]
+        channel_configs, channel_count = get_channel_configs()
+        LOG.info('Found %d channels', channel_count)
+        for i in range(channel_count):
+            channel_config = channel_configs[i]
             LOG.info('Channel index %d: %s',
                      channel_config.channelIndex,
                      channel_config.name.decode('ascii'))
             configs.append({'interface': 'vector',
                             'app_name': None,
                             'channel': channel_config.channelIndex})
         return configs
+
+def get_channel_configs():
+    if vxlapi is None:
+        return [], 0
+    driver_config = vxlapi.XLdriverConfig()
+    try:
+        vxlapi.xlOpenDriver()
+        vxlapi.xlGetDriverConfig(driver_config)
+        vxlapi.xlCloseDriver()
+    except:
+        pass
+    return driver_config.channel, driver_config.channelCount
```

Comment:
A nicer way to do this to avoid even more code later is something like `return [driver_config.channel[i] for i in range(driver_config.channelCount)`. But it's really a small detail. :)

Reply:
I like that one!